src/coreclr/jit/morph.cpp

```diff
@@ -9916,16 +9916,6 @@ GenTree* Compiler::fgOptimizeHWIntrinsic(GenTreeHWIntrinsic* node)
                 break;
             }
 
-#if defined(TARGET_XARCH)
-            if ((simdSize == 8) && !compOpportunisticallyDependsOn(InstructionSet_SSE41))
-            {
-                // When SSE4.1 isn't supported then Vector2 only needs a single horizontal add
-                // which means the result isn't broadcast across the entire vector and we can't
-                // optimize
-                break;
-            }
-#endif // TARGET_XARCH
```

Comment:
This code was incorrect, as we expect that broadcast should always be occurring.

The lowering logic was doing the right thing already for SSE2 and SSE41+ machines, but not for SSE3 or SSSE3 machines where `HorizontalAdd` exists